Fix JSON Jackson jacksonConversionPojo test-addressed feedback #2726

ffang opened the pull request:
(cherry picked from commit 751063196f3a99e07906cf69b512f6ab970392e7)

<!-- Uncomment and fill this section if your PR is not trivial
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html
-->

ppalaga commented on extensions/jackson/deployment/src/test/java/org/apache/camel/quarkus/component/jackson/JacksonTypeConverterPojoTest.java:
```diff
@@ -0,0 +1,83 @@
+
+        Properties props = new Properties();
+        props.setProperty("camel.context.name",
+                "camel-quarkus-integration-tests-dataformats-json-jackson-type-converter-pojo");
```
Is there any practical reason to set the context name? If not, we could save 15 lines of code removing the whole `applicationProperties()` method.

ffang replied:
right, revised in all three classes

ppalaga commented on extensions/jackson/deployment/src/test/java/org/apache/camel/quarkus/component/jackson/JacksonTypeConverterSimpleTest.java:
```diff
@@ -0,0 +1,97 @@
+
+    @Test
+    public void jacksonConversionSimple() throws Exception {
+        context.getGlobalOptions().put(JacksonConstants.ENABLE_TYPE_CONVERTER, "true");
```
This line is redundant given that so the same in `Routes.configure()` below, isn't it?

ffang replied:
yes, revised accordingly

ppalaga commented on extensions/jackson/deployment/src/test/java/org/apache/camel/quarkus/component/jackson/JacksonTypeConverterSimpleTest.java:
```diff
@@ -0,0 +1,97 @@
+
+        Properties props = new Properties();
+        props.setProperty("camel.context.name",
+                "camel-quarkus-integration-tests-dataformats-json-jackson-type-converter-simple");
```
Same as above: do we need `applicationProperties()` at all?

ppalaga commented on extensions/jacksonxml/deployment/src/test/java/org/apache/camel/quarkus/component/jacksonxml/deployment/JacksonxmlTypeConverterTest.java:
```diff
@@ -0,0 +1,92 @@
+    @Test
+    public void jacksonxmlConversion() throws Exception {
+        Exchange exchange = new DefaultExchange(context);
+        context.getGlobalOptions().put(JacksonXMLConstants.ENABLE_TYPE_CONVERTER, "true");
```
Redundant line because we do the same in Route.configure() below?

ppalaga commented on integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/jackson/json/JacksonJsonResource.java:
```diff
@@ -493,32 +490,19 @@ public void jacksonUnmarshalListJackson(String body) throws Exception {
 
-            String json = (String) producerTemplate.requestBody("direct:jackson-conversion-pojo-test", order);
-            assertEquals("{\"id\":0,\"partName\":\"Camel\",\"amount\":1,\"customerName\":\"Acme\"}", json);
+            //context.getGlobalOptions().put(JacksonConstants.ENABLE_TYPE_CONVERTER, "true");
+            //context.getGlobalOptions().put(JacksonConstants.TYPE_CONVERTER_TO_POJO, "true");
```
```suggestion
```

ppalaga commented on integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/jackson/json/JacksonJsonResource.java:
```diff
@@ -493,32 +490,19 @@ public void jacksonUnmarshalListJackson(String body) throws Exception {
 
-            context.getGlobalOptions().put(JacksonConstants.TYPE_CONVERTER_MODULE_CLASS_NAMES,
-                    JaxbAnnotationModule.class.getName());
+            //context.getGlobalOptions().put(JacksonConstants.TYPE_CONVERTER_MODULE_CLASS_NAMES,
+            //      JaxbAnnotationModule.class.getName());
```
```suggestion
```

ffang replied:
done

ppalaga reviewed (commented):
Thanks @ffang, looks quite good, some questions and comments inline

ffang commented:
Hi @ppalaga ,

Thanks for the feedback, just pushed a new commit
https://github.com/apache/camel-quarkus/pull/2790/commits/240a0b773c56448a9deb8987e59fd56d905b9777
And addressed all your suggestions accordingly.

Freeman

aldettinger commented on extensions/jackson/deployment/src/test/java/org/apache/camel/quarkus/component/jackson/JacksonTypeConverterPojoTest.java:
```diff
@@ -0,0 +1,61 @@
+                    .addClass(Routes.class)
+                    .addClass(Order.class));
+
+    private static ProducerTemplate producerTemplate;
```
I think it should be possible pass the ProducerTemplate by injection so that the field is not static, like:
@Inject
ProducerTemplate producerTemplate

aldettinger commented on extensions/jackson/deployment/src/test/java/org/apache/camel/quarkus/component/jackson/JacksonTypeConverterSimpleTest.java:
```diff
@@ -0,0 +1,74 @@
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(Routes.class)
+                    .addClass(Order.class));
+    private static CamelContext context;
```
Should work here too:
@Inject
CamelContext context;

aldettinger reviewed (approved):
@ffang I like the idea that it's like testing a deployment. Then the route configure() method really looks like what a camel-quarkus developer would do and the itests on their side are more isolated/maintainable.

As commented above, I think it should be possible to get rid of the static fields in deployment test. Beyond that, I would say we are good to go. Well done :+1:

ffang commented:
> @ffang I like the idea that it's like testing a deployment. Then the route configure() method really looks like what a camel-quarkus developer would do and the itests on their side are more isolated/maintainable.
> 
> As commented above, I think it should be possible to get rid of the static fields in deployment test. Beyond that, I would say we are good to go. Well done 👍

Thanks @aldettinger for the nice suggestion!

Revised here accordingly https://github.com/apache/camel-quarkus/pull/2790/commits/f7e25b0c6415b5b0c19333552ca0f8ffefe30084
Freeman

aldettinger commented:
@ffang Not sure it has been squashed correctly, I still see 7 commits in the PR. The "Files changed" aggregated diff looks good. In this situation, I would do sometginf like `git rebase -i HEAD~7`, fixups, rebase on main and push -f to the PR... I'm sure there are other way to do thought :)

aldettinger commented:
Yeah looks good. Let's wait the build and rebase/merge the PR then :+1:

ffang commented:
The CI build is full green, so I merged.

Thanks all for the good input!
